disneyland/storage.go

```diff
@@ -172,45 +191,73 @@ func (storage *DisneylandStorage) UpdateJob(job *Job) (*Job, error) {
 	return resultJob, err
 }
 
-func (storage *DisneylandStorage) PullJobs(how_many uint32, kind string) ([]*Job, error) {
+func (storage *DisneylandStorage) PullJobs(how_many uint32, project string, kind string) (*ListOfJobs, error) {
 	tx, err := storage.db.Begin()
 	if err != nil {
 		return nil, err
 	}
-	resultJobs := []*Job{}
-	sqlStr := `
-	WITH updatedPts as (
-		WITH pulledPts as (
-			SELECT id, project, kind
+	strQuery := `SELECT id, project, kind
 			FROM jobs
-			WHERE status=$1 and kind=$2
-			LIMIT $3
+			WHERE status=$1`
+	var rows *sql.Rows
+
+	if kind != "" {
+		strQuery += ` AND kind=$2`
+		if project != "" {
+			strQuery += ` AND project=$3`
+			strQuery = `	WITH updatedPts AS
```

Comment:
`=` after `+=`. Method became too monstrous, I recommend splitting it/removing nesting.